### ISSUE TITLE
Redmine crashed on Postgres request

### ISSUE THREAD

**Issue by dmitrymaximov** [dhanasingh/redmine_wktime#17]
Redmine crashed on Postgres request

Logs below:

Started GET "/" for 95.211.89.207 at 2015-05-14 03:45:36 -0400
Processing by WelcomeController#index as HTML
 Current user: supportex.net (id=57)
 Rendered news/_news.html.erb (7.7ms)
 Rendered welcome/index.html.erb within layouts/base (17.0ms)
PG::UndefinedTable: ERROR:  missing FROM-clause entry for table "time_entries"
LINE 1: SELECT "projects".* FROM "projects" WHERE ((time_entries.ord...
                                                   ^
: SELECT "projects".* FROM "projects" WHERE ((time_entries.order_id IN ('')) OR (time_entries.user_id = 57 AND time_entries.
order_id IN ('')) OR ((projects.status <> 9 AND projects.id IN (SELECT em.project_id FROM enabled_modules em WHERE em.name='
time_tracking')) AND (projects.id IN (12,27,50,2,26,28,55) OR projects.id IN (12,27,54,85) OR projects.id IN (12,39) OR proj
ects.id IN (44,47))))
Completed 500 Internal Server Error in 80ms (ActiveRecord: 22.3ms)

ActionView::Template::Error (PG::UndefinedTable: ERROR:  missing FROM-clause entry for table "time_entries"
LINE 1: SELECT "projects".* FROM "projects" WHERE ((time_entries.ord...
                                                   ^
: SELECT "projects".* FROM "projects" WHERE ((time_entries.order_id IN ('')) OR (time_entries.user_id = 57 AND time_entries.
order_id IN ('')) OR ((projects.status <> 9 AND projects.id IN (SELECT em.project_id FROM enabled_modules em WHERE em.name='
time_tracking')) AND (projects.id IN (12,27,50,2,26,28,55) OR projects.id IN (12,27,54,85) OR projects.id IN (12,39) OR proj
ects.id IN (44,47))))):
   22: <div id="top-menu">
   23:     <div id="account">
   24:         <%= render_menu :account_menu -%>
   25:     </div>
   26:     <%= content_tag('div', "#{l(:label_logged_as)} #{link_to_user(User.current, :format => :username)}".html_safe, :
id => 'loggedas') if User.current.logged? %>
   27:     <%= render_menu :top_menu if User.current.logged? || !Setting.login_required? -%>
   28: </div>
 lib/redmine/menu_manager.rb:446:in `call'
 lib/redmine/menu_manager.rb:446:in `allowed?'
 lib/redmine/menu_manager.rb:164:in `block in menu_items_for'
 lib/redmine/menu_manager.rb:163:in `each'
 lib/redmine/menu_manager.rb:163:in `menu_items_for'
 lib/redmine/menu_manager.rb:90:in `render_menu'
 app/views/layouts/base.html.erb:25:in `_app_views_layouts_base_html_erb___1517620741803361580_43006300'

**Comment by dmitrymaximov** [comment]
Environment:
  Redmine version                3.0.2.stable
  Ruby version                   2.2.1-p85 (2015-02-26) [x86_64-linux]
  Rails version                  4.2.1
  Environment                    production
  Database adapter               PostgreSQL

**Comment by suganya-thulasiraman** [comment]
What version of Time & Expense you use?
Time & Expense 1.8 is compatible with Redmine 3.0.x, older versions are not compatible with Redmine 3.0.x

**Comment by dmitrymaximov** [comment]
I have reinstalled plugin to 1.8.1 version. The new issue when I open https://<my-redmine>.com/wktime/index:

Current user: dmitry.maximov (id=73)
PG::InvalidColumnReference: ERROR:  for SELECT DISTINCT, ORDER BY expressions must appear in select list
LINE 1: ... (users.type='User' AND users.status=1)  ORDER BY users.firs...
                                                             ^
: SELECT DISTINCT "members".* FROM "members" INNER JOIN "users" ON "users"."id" = "members"."user_id" INNER JOIN "member_roles" ON "member_roles"."member_id" = "members"."id" INNER JOIN "roles" ON "roles"."id" = "member_roles"."role_id" WHERE "members"."project_id" = $1 AND (users.type='User' AND users.status=1)  ORDER BY users.firstname ASC,users.lastname ASC
Completed 500 Internal Server Error in 29ms (ActiveRecord: 10.1ms)

ActiveRecord::StatementInvalid (PG::InvalidColumnReference: ERROR:  for SELECT DISTINCT, ORDER BY expressions must appear in select list
LINE 1: ... (users.type='User' AND users.status=1)  ORDER BY users.firs...
                                                             ^
: SELECT DISTINCT "members".* FROM "members" INNER JOIN "users" ON "users"."id" = "members"."user_id" INNER JOIN "member_roles" ON "member_roles"."member_id" = "members"."id" INNER JOIN "roles" ON "roles"."id" = "member_roles"."role_id" WHERE "members"."project_id" = $1 AND (users.type='User' AND users.status=1)  ORDER BY users.firstname ASC,users.lastname ASC):
  activerecord (4.2.1) lib/active_record/connection_adapters/postgresql_adapter.rb:637:in `prepare'

**Comment by dmitrymaximov** [comment]
Any news for this issue?

**Comment by suganya-thulasiraman** [comment]
we have fixed it, but it will be part of 1.8.2

Here are the methods which should be replaced in wktime_controller.rb to fix this issue

```
def setMembers		
	@groups = Group.sorted.all
	@members = Array.new
	if !params[:tab].blank? && params[:tab] =='wkexpense'
		filter_type = session[:wkexpense][:filter_type]
		project_id = session[:wkexpense][:project_id]
	else
		filter_type = session[:wktimes][:filter_type]
		project_id = session[:wktimes][:project_id]
	end
	hookMem = call_hook(:controller_get_member, { :filter_type => filter_type})
	if filter_type == '1' || (hookMem.blank? && filter_type !='2')
		hookProjMem = call_hook(:controller_project_member, {  :project_id => project_id})
		if !hookProjMem.blank?
			projMem = hookProjMem[0].blank? ? [] : hookProjMem[0]
		else
			projMem = @selected_project.members.order("#{User.table_name}.firstname ASC,#{User.table_name}.lastname ASC")
		end				
		@members = projMem.collect{|m| [ m.name, m.user_id ] }
	elsif filter_type == '2'
		userList = []
		userList = getGrpMembers			
		userList.each do |users|
			@members << [users.name,users.id.to_s()]
		end		
	else			
		if !hookMem.blank?
			@members = hookMem[0].blank? ? @members : hookMem[0]		
		end
	end
	@members = @members.uniq
end

```

```
def getusers
	project = Project.find(params[:project_id])
	userStr = ""
	userList = call_hook(:controller_project_member,{ :project_id => params[:project_id]})
	if !userList.blank?
		projmembers = userList[0].blank? ? nil : userList[0]
	else
		projmembers = project.members.order("#{User.table_name}.firstname ASC,#{User.table_name}.lastname ASC")
		end
	projmembers = projmembers.to_a.uniq if !projmembers.nil?
	projmembers.each do |m|
		userStr << m.user_id.to_s() + ',' + m.name + "\n"
	end	
	respond_to do |format|
		format.text  { render :text => userStr }
	end
end

```

**Comment by dmitrymaximov** [comment]
Great, thank you. Do you have a timeline for 1.8.2 version?

**Comment by suganya-thulasiraman** [comment]
Fixed in 1.9
